controllers/barbicanapi_controller.go

```diff
@@ -197,15 +198,19 @@ func (r *BarbicanAPIReconciler) getSecret(
 	secretName string,
 	envVars *map[string]env.Setter,
 ) (ctrl.Result, error) {
-	hash, result, err := secret.VerifySecret(ctx, types.NamespacedName{Name: instance.Spec.Secret, Namespace: instance.Namespace}, []string{"AdminPassword"}, h.GetClient(), time.Second*10)
+	expectedField := "transport_url"
+	if strings.HasSuffix(secretName, "osp-secret") {
```

Comment:
We don't want to hard-code `"osp-secret"` here, because that value is configurable in the API spec.  This should be updated to use the value provided in the spec `instance.Spec.Secret`, which already defaults to `"osp-secret"`:

https://github.com/openstack-k8s-operators/barbican-operator/blob/main/api/v1beta1/common_types.go#L40

Reply:
wouldn't it be easier to maintain to have another new parameter to the func where the caller passes in the `[]string{}` to check in `VerifySecret()`. So for the `instance.Spec.Secret` it would be`[]string{instance.Spec.PasswordSelectors.Service}` and for the transportURL `[]string{"transport_url"}` . I think this would make it easier to understand and won't require this condition

Reply:
@stuggi that's the same suggestion I made in the other comment below. :point_down:

Reply:
oh right, sorry haven't read it close enough